File: OOP-3/Lab2/Linear.java
import java.util.*;
class Linear
{
	public void linearSearch(int a[], int n)
	{
	    int flag=0;
	    Scanner scL=new Scanner(System.in);
	    System.out.println("Enter element to be searched.");
	    int x=scL.nextInt();
        for(int i=0; i<n; i++)
        {
            if(a[i]==x)
            {
                flag=1;
                System.out.println("Element found at position "+(i+1));
                break;
            }
        }
        if(flag==0)
        System.out.println("Element not found.");
	}
	public static void main(String args[])
	{
	    Scanner sc=new Scanner(System.in);
	    System.out.println("Enter array size.");
	    int n=sc.nextInt();
        int a[]=new int[n];
        System.out.println("Enter "+n+" elements.");
        for(int i=0; i<n; i++)
        a[i]=sc.nextInt();
        Linear l=new Linear();
        l.linearSearch(a, n);
	}
}

File: DSL-3/Lab1/Linear.c
#include<stdio.h>
int i;
int Lsearch(int a[], int n, int x)
{
	int flag=0,pos;
	for(i=0; i<n; i++)
	{
	    if(a[i]==x)
	    {
	        pos=i+1;
	        flag=1;
	        break;
	    }
	}
	if(flag==0)
	printf("Element not found.\n");
	else
	printf("Element found at position %d.", pos);
}
int main()
{
    int n, x;
	printf("Enter size of array.\n");
	scanf("%d", &n);
	int a[n];
	printf("Enter %d elements.\n", n);
	for(i=0; i<n; i++)
	    scanf("%d", &a[i]);
	printf("Enter element to be searched.\n");
	scanf("%d", &x);
	Lsearch(a, n, x);    
}

